internals/feature_links.py

```diff
@@ -14,12 +14,19 @@
 # limitations under the License.
 
 import logging
+import datetime
+from threading import Semaphore
+from concurrent.futures import ThreadPoolExecutor
 from typing import Any
 from internals.core_models import FeatureEntry
 from internals.link_helpers import Link
 
 from google.cloud import ndb  # type: ignore
 
+executor = ThreadPoolExecutor()
```

Comment:
I was thinking that you would use Google Cloud Tasks to enqueue a request that would run a separate handler.  There are examples of doing this in our code.   E.g.,
https://github.com/GoogleChrome/chromium-dashboard/blob/main/internals/notifier.py#L534

The advantage of using Google Cloud Tasks is that each task execution is a separate request that can be found more easily in the logs, if the task handler fails it is automatically retried, and there is an admin page in the Google Cloud Console that shows the pending task queue.

Reply:
I believe that Cloud Tasks should run the task in under one second. 

However, the indexing process will need to wait for the requests for each URL to finish, which could take several seconds.

So, maybe the time between the initial API call and the retry should be more like 10 seconds instead of 2 seconds.